src/parser/jobs/ast/modules/CelestialOpposition.tsx

```diff
@@ -47,27 +64,27 @@ export default class CelestialOpposition extends Module {
 		this.lastUse = event.timestamp
 	}
 
-	private onSect(event: BuffEvent) {
-		this.activeSect = getDataBy(this.data.statuses, 'id', event.ability.guid)
+	private onSect(event: Events['statusApply']) {
+		this.activeSect = getDataBy(this.data.statuses, 'id', event.status)
 	}
 
 	onComplete() {
 		const holdDuration = this.uses === 0 ? this.parser.currentDuration : this.totalHeld
-		const usesMissed = Math.floor(holdDuration / this.data.actions.CELESTIAL_OPPOSITION.cooldown)
-		const fightDuration = this.parser.fight.end_time - this.parser.fight.start_time
-		const maxUses = (fightDuration / this.data.actions.CELESTIAL_OPPOSITION.cooldown) - 1
+		const usesMissed = Math.floor(holdDuration / (this.data.actions.CELESTIAL_INTERSECTION.cooldown))
+		//TODO update max uses to not use the whole duration such as when there are no available targets
+		const maxUses = (this.parser.pull.duration / this.data.actions.CELESTIAL_INTERSECTION.cooldown) - 1
```

Comment:
updated variable names based on feedback above
in terms of the TODO, yeah, I'll definitely have to discuss with others or come up with a better approach than the current one
since this feedback would be the same for another todo, I'll keep it in mind for now and possibly take a stab at it in EW